WP 6.1 Compatibility: Fix the giant icons in the `DatePicker` component

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Closes #1736 

- ~Fix that the style of `VisuallyHidden` component is not working since WP 5.9.~
- Fix the giant icons in the `DatePicker` component
   ![image](https://user-images.githubusercontent.com/17420811/200817938-211d7120-146d-4abe-9afe-9e65d0143fb9.png)

This PR also implements the base of the app shell (#1548):

- ~Add a new value of `layoutName` to `useLayout` hook for isolating GLA's style fixes.~
- Wrap all top-level admin pages with an app shell and CSS class `gla-admin-page`.
- Add CSS class `gla-admin-page` to the root of `Modal` as well.

### Screenshots:

![image](https://user-images.githubusercontent.com/17420811/201340263-2afb0fb9-f767-4b68-a2dd-36cef2a7ca6e.png)

### Detailed test instructions:

~1. Install a WP version >= 5.9~
~2. Set up paid ads or go to GLA Settings page to view the external link on the card UI of the connected Google Ads account.~
~3. The text "(opens in a new tab)" within the external link should be invisible.~

1. Install WP 6.1.
2. Go to the Dashboard page.
3. Click the date range option to open datepicker. Switch to custom tab to verify if the icon size of navigation buttons is appropriate.

### Changelog entry

> Fix - WordPress 6.1 Compatibility: The size of navigation icons in Datepicker component should not be a giant size.

eason9487 committed:
Fix that the style of `VisuallyHidden` component is not working since WP 5.9

eason9487 committed:
Add a new value of `layoutName` to useLayout for isolating GLA's style fixes

eason9487 committed:
Wrap all top-level admin pages with an app shell

tomalec reviewed (approved):
Tested locally, inspected the code, LGTM

tomalec commented:
I really like the introduction of the app shell concept.

:grey_question: 
However, I think it would be nice to specify what we consider the "app shell", as I'd expect it to be an actual element enclosing the app's HTML subtree, or something narrowing the scope of elements to the ones created by our extension.

Whereas if you visit `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings` and do `document.querySelectorAll('.gla-admin-page button')` you will get buttons from WP side menu, `#activity-panel-tab-activity` from WooCommerce chrome, etc. which are beyond our control.

Is this really what we wanted? Why don't we just wrap every GLA's page with `<gla-page>`/`<div class="gla-page">` or attach this class to the enclosing container/direct `.parentElement`?

tomalec commented:
Made an alternative fix in https://github.com/woocommerce/google-listings-and-ads/pull/1762

eason9487 committed:
Move CSS  class gla-admin-page to nodes that are closer to the root of GLA pages in the DOM tree.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1739#issuecomment-1309205776

eason9487 committed:
Revert "Add a new value of `layoutName` to useLayout for isolating GLA's style fixes"

This reverts commit 091b75f9cab1a794bafad8c1228db47a31f1c8f2.

eason9487 commented:
@tomalec, thanks for the review.

> However, I think it would be nice to specify what we consider the "app shell", as I'd expect it to be an actual element enclosing the app's HTML subtree, or something narrowing the scope of elements to the ones created by our extension.

> Is this really what we wanted? Why don't we just wrap every GLA's page with `<gla-page>`/`<div class="gla-page">` or attach this class to the enclosing container/direct `.parentElement`?

Since some components are created through ReactDOM.createPortal, they are usually mounted directly under the `<body>`, but do not provide identifiers for precise scoping, such as `Modal`, so I chose to add `.gla-full-page` to `<body>` to ensure those components would also apply CSS fixes. But as you mentioned, it's indeed not ideal.

I searched the GLA codebase and the only components that use `ReactDOM.createPortal` are `Modal` and `Snackbar` (used through `createNotice`). The former can be scoping CSS class in `AppModal`, the latter probably doesn't need to be adjusted at the moment, so I made some adjustments to scope GLA admin pages more accurately.

- https://github.com/woocommerce/google-listings-and-ads/pull/1739/commits/9decd2908953aa6a11a5f7315237987e49690f4e : Move CSS class `gla-admin-page` to nodes that are closer to the root of GLA pages in the DOM tree.
- https://github.com/woocommerce/google-listings-and-ads/pull/1739/commits/52ef95f5a0a237ee41d20a061e019476779324c2 : Revert the change in the `useLayout` hook.

eason9487 committed:
Replace the fix of `VisuallyHidden` with the one for `DatePicker`

Ref: https://github.com/woocommerce/google-listings-and-ads/pull/1762#issuecomment-1311437832

eason9487 committed:
Rebalance the file size limit of js/build/index.css to 12.83 * 1.2 ≈ 15.4

eason9487 commented:
@tomalec, thanks for proposing [a nicer solution](https://github.com/woocommerce/google-listings-and-ads/pull/1739#issuecomment-1310850630).

I adjusted this PR as per https://github.com/woocommerce/google-listings-and-ads/pull/1762#issuecomment-1311437832 by https://github.com/woocommerce/google-listings-and-ads/pull/1739/commits/3293fb01d198cd69bb6ed03454a0707891dba963 and https://github.com/woocommerce/google-listings-and-ads/pull/1739/commits/aecd97f4124bc626e8f03eea26aa539a6df82613. PR description was also updated. Could you help with a new round of code reviews?

tomalec reviewed (approved):
Thanks for the research and explanations. Regarding `createNotice` I think it's better we try to avoid tempering its styles to sustain consistent experience for merchants, so leaving it without `gla-admin-page` scope seems fine to me.

Tested locally, reviewed the code LGTM
